rasa/core/policies/rule_policy.py

```diff
@@ -664,12 +663,15 @@ def _is_rule_applicable(
         return bool(
             # rule is shorter than current turn index
             turn_index >= len(reversed_rule_states)
-            # current rule and state turns are empty
-            or (not reversed_rule_states[turn_index] and not conversation_state)
+            # current rule and state turns are conversation starters
```

Comment:
I think the comment is slightly useful, but we could make it more useful by stating something like:
```suggestion
            # current rule and state turns are conversation starters.
            # Don't consider any other state than actions in this case.
```

Reply:
I'm re-writing the comments + adding some more :-)